modules/local/counts/calculate_tpm/templates/calculate_tpm.py

```diff
@@ -34,7 +34,7 @@ def format_yaml_like(data: dict, indent: int = 0) -> str:
 df_lengths = df_lengths / 1e3
 df_lengths = df_lengths.groupby(df_lengths.index).mean()
 
-df_lengths = df_lengths.loc[df_counts.index]
+df_lengths = df_lengths.loc[df_lengths.index.isin(df_counts.index)]
```

Comment:
Did you check if it can happen that the dataframes have different orders? 

If we cannot be entirely sure of this, it might be better to build the intersection of both indices and then subset both to the intersection